meta-lmp-base/recipes-support/compose-apps-early-start/compose-apps-early-start/compose-apps-early-start-recovery

```diff
@@ -1,15 +1,31 @@
 #!/bin/sh
 
-# Force compose down, restart docker and try again
+function recover_compose_apps() {
+	# Force compose down, restart docker and try again
 
-for app in `ls /var/sota/compose-apps` ; do
-	cd /var/sota/compose-apps/${app}
-	docker compose down
-done
+	for app in `ls /var/sota/compose-apps` ; do
+		cd /var/sota/compose-apps/${app}
+		docker compose down
+	done
 
-# TODO: extend with support for restorable-apps, which should
-# allow us to remove /var/lib/docker completely
+	systemctl restart docker
+	systemctl reset-failed compose-apps-early-start.service
+	systemctl restart compose-apps-early-start.service
+}
 
-systemctl restart docker
-systemctl reset-failed compose-apps-early-start.service
-systemctl restart compose-apps-early-start.service
+function recover_restorable_apps() {
+	systemctl stop docker
+	rm -rf /var/sota/compose-apps/*
+	rm -rf /var/lib/docker
+
+	systemctl restart compose-apps-register-images.service
+	systemctl start docker
+	systemctl reset-failed compose-apps-early-start.service
+	systemctl restart compose-apps-early-start.service
```

Comment:
reset-failed and restart is common to both, can be done at the end of the script.